openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -21,450 +21,428 @@
+            if key_value_pair["name"] == prop_name
+        ]
+        if prop_list:
+            return prop_list[0].get("value")
+
+    def get_value_by_names(self, key_value_list: list, prop_names: list):
+        """Gets the first available value for one of the `prop_names`
+        property names"""
+        for prop_name in prop_names:
+            if val := self.get_value_by_name(key_value_list, prop_name):
+                return val
+
+    def extract_media_type(self, media_info):
+        media_type = media_info.get("mediatype")
+
+        if media_type in IMAGE_MEDIATYPES:
+            return IMAGE
+        elif media_type in AUDIO_MEDIATYPES:
+            return AUDIO
+
+        logger.info(
+            f"Incorrect mediatype: {media_type} not in "
             f"valid mediatypes ({IMAGE_MEDIATYPES}, {AUDIO_MEDIATYPES})"
         )
         return None
 
+    def extract_audio_category(self, parsed_data):
+        """Set category to "pronunciation" for any audio with
+        pronunciation of a word or a phrase"""
+        for category in parsed_data["meta_data"].get("categories", []):
+            if "pronunciation" in category.lower():
+                return "pronunciation"
+
+    def extract_title(self, media_info):
+        # Titles often have 'File:filename.jpg' form
+        # We remove the 'File:' and extension from title
+        title = self.get_ext_value(media_info, "ObjectName")
+        if title is None:
+            title = media_info.get("title")
+        if title.startswith("File:"):
+            title = title.replace("File:", "", 1)
+        last_dot_position = title.rfind(".")
+        if last_dot_position > 0:
+            possible_extension = title[last_dot_position:]
+            if possible_extension.lower() in {".png", ".jpg", ".jpeg", ".ogg", ".wav"}:
+                title = title[:last_dot_position]
+        return title
+
+    def extract_date_info(self, media_info):
+        date_originally_created = self.get_ext_value(media_info, "DateTimeOriginal")
+        last_modified_at_source = self.get_ext_value(media_info, "DateTime")
+        return date_originally_created, last_modified_at_source
+
+    def extract_creator_info(self, media_info):
+        artist_string = self.get_ext_value(media_info, "Artist")
+
+        if not artist_string:
+            return None, None
+
+        artist_elem = html.fromstring(artist_string)
+        # We take all text to replicate what is shown on Wikimedia Commons
+        artist_text = "".join(artist_elem.xpath("//text()")).strip()
+        url_list = list(artist_elem.iterlinks())
+        artist_url = self.cleanse_url(url_list[0][2]) if url_list else None
+        return artist_text, artist_url
+
+    def extract_category_info(self, media_info):
+        categories_string = self.get_ext_value(media_info, "Categories") or ""
+
+        categories_list = categories_string.split("|")
+        return categories_list
+
+    def extract_file_type(self, media_info):
```

Comment:
File extraction from the media URL field is done in the MediaStore if the `filetype` is None, so it's better to remove this ad-hoc function.

Reply:
I think we need this because for wikimedia we later do some extra handling on `filetype` to set it to [illustration for svgs](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py#L333).

Reply:
We were previously automatically setting all of the SVG files to `illustration`, so it makes sense to move this to the `ImageStore` class, and categorize the images after we extract their file types.

Reply:
To clarify -- you're saying that we can set media type to `illustration` for _all_ svg files regardless of provider? I'm wondering if it would be worth making that change separately from the refactor if it will potentially have a broader effect.

Reply:
#716 opened to track this.